sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSource.java

```diff
@@ -331,7 +331,13 @@ public void populateDisplayData(DisplayData.Builder builder) {
       try {
         checkState(fileOrPatternSpec.isAccessible(),
                    "Bundle splitting should only happen at execution time.");
-        for (final String file : FileBasedSource.expandFilePattern(fileOrPatternSpec.get())) {
+        Collection<String> expandedFiles =
+            FileBasedSource.expandFilePattern(fileOrPatternSpec.get());
+        if (expandedFiles.isEmpty()) {
```

Comment:
Done.

This method is marked final so there's no worry about overriding classes' behavior.